testdata/testdata.go

```diff
@@ -262,6 +67,185 @@ func GenTestData(sc *arrow.Schema, opts GenTestDataOptions) []arrow.Record {
 	return records
 }
 
+func getExampleJSON(colName string, dataType arrow.DataType, opts GenTestDataOptions) string {
+	// handle lists (including maps)
+	if arrow.IsListLike(dataType.ID()) {
+		if dataType.ID() == arrow.MAP {
+			k := getExampleJSON(colName, dataType.(*arrow.MapType).KeyType(), opts)
+			v := getExampleJSON(colName, dataType.(*arrow.MapType).ValueType().Field(1).Type, opts)
+			return fmt.Sprintf(`[{"key": %s,"value": %s}]`, k, v)
+		}
+		inner := dataType.(*arrow.ListType).Elem()
+		return `[` + getExampleJSON(colName, inner, opts) + `]`
+	}
+	// handle extension types
+	if arrow.TypeEqual(dataType, types.ExtensionTypes.UUID) {
+		u := uuid.New()
+		if opts.StableUUID != uuid.Nil {
+			u = opts.StableUUID
+		}
+		return `"` + u.String() + `"`
+	}
+	if arrow.TypeEqual(dataType, types.ExtensionTypes.JSON) {
+		return `"{\"test\":\"test\"}"`
+	}
+	if arrow.TypeEqual(dataType, types.ExtensionTypes.Inet) {
+		return `"192.0.2.0/24"`
+	}
+	if arrow.TypeEqual(dataType, types.ExtensionTypes.Mac) {
+		return `"aa:bb:cc:dd:ee:ff"`
+	}
+
+	// handle integers
+	if arrow.IsInteger(dataType.ID()) {
+		return "-1"
+	}
+
+	// handle unsigned integers
+	if arrow.IsUnsignedInteger(dataType.ID()) {
+		return "1"
+	}
+
+	// handle floats
+	if arrow.IsFloating(dataType.ID()) {
+		return "1.1"
+	}
+
+	// handle decimals
+	if arrow.IsDecimal(dataType.ID()) {
+		return "1.1"
+	}
+
+	// handle booleans
+	if arrow.TypeEqual(dataType, arrow.FixedWidthTypes.Boolean) {
+		return "true"
+	}
+
+	// handle strings
+	stringTypes := []arrow.DataType{
+		arrow.BinaryTypes.String,
+		arrow.BinaryTypes.LargeString,
+	}
+	for _, stringType := range stringTypes {
+		if arrow.TypeEqual(dataType, stringType) {
+			if colName == schema.CqSourceNameColumn.Name {
+				return `"` + opts.SourceName + `"`
+			}
+			return `"AString"`
+		}
+	}
+
+	// handle binary types
+	binaryTypes := []arrow.DataType{
+		arrow.BinaryTypes.Binary,
+		arrow.BinaryTypes.LargeBinary,
+	}
+	for _, binaryType := range binaryTypes {
+		if arrow.TypeEqual(dataType, binaryType) {
+			return `"AQIDBA=="` // base64 encoded 0x01, 0x02, 0x03, 0x04
```

Comment:
Explicitly run base64 encode instead of magic strings

Reply:
There is a comment that says what it is, so flipping the order of comment / string doesn't really matter imo. For me this is useful to show that the base64-encoded string uses padding